### PR TITLE
Feature add three six and nine months options backend

### PR DESCRIPTION
**Pull request by c-andy-candies**
I adapted the code from #3161 to add the sorting options for 3, 6 and 9 months which I think would be very cool to have, as someone pointed out here #3217 

As the original PR I also am currently doing PRs for frontend and translations.

LemmyNet/lemmy-ui#1420

LemmyNet/lemmy-translations#69

This is my first contribution so please forgive me if I did something wrong.

### CODE REVIEW

**Comment by Nutomic** [comment]
Seems fine, but we shouldnt add any more after this. It will already result in a very long list.

**Review by dessalines — APPROVED** [review]
LGTM, thanks!

File: migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-- update the default sort type
+update local_user set default_sort_type = 'TopDay' where default_sort_type in ('TopThreeMonths', 'TopSixMonths', 'TopNineMonths');
+
+-- rename the old enum
+alter type sort_type_enum rename to sort_type_enum__;
+-- create the new enum
+CREATE TYPE sort_type_enum AS ENUM ('Active', 'Hot', 'New', 'Old', 'TopDay', 'TopWeek', 'TopMonth', 'TopYear', 'TopAll', 'MostComments', 'NewComments');
```
Comment (L3v3L) [inline migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql:7]: this should include the TopHour, TopSixHour and TopTwelveHour enums

**Comment by L3v3L** [comment]
I think the migration needs to be timestamp after an already released migration or it wont run.
@dessalines

File: migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql (hunk 1 above)
Comment (c-andy-candies) [reply migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql:7]: You're right, I was not sure when I made the PR

Comment (L3v3L) [reply migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql:7]: usually there's no take backsies with migrations, but I think in this case it's a down migration and is probably fine just opening a new PR with the hotfix of this file.

Comment (c-andy-candies) [reply migrations/2023-06-20-191145_add_listingtype_sorttype_3_6_9_months_enums/down.sql:7]: Opened a PR #3370

**Pull request merged**